server/src/server.ts

```diff
@@ -93,52 +91,10 @@ connection.onDidChangeWatchedFiles((change) => {
   bitBakeProjectScanner.rescanProject()
 })
 
-connection.onCompletion((textDocumentPositionParams: TextDocumentPositionParams): CompletionItem[] => {
-  logger.debug('onCompletion')
-  const documentAsText = documentAsTextMap.get(textDocumentPositionParams.textDocument.uri)
-  if (documentAsText === undefined) {
-    return []
-  }
-
-  const word = analyzer.wordAtPointFromTextPosition({
-    ...textDocumentPositionParams,
-    position: {
-      line: textDocumentPositionParams.position.line,
-      // Go one character back to get completion on the current word. This is used as a parameter in descendantForPosition()
-      character: Math.max(textDocumentPositionParams.position.character - 1, 0)
-    }
-  })
-
-  logger.debug(`onCompletion - current word: ${word}`)
-
-  let symbolCompletions: CompletionItem[] = []
-  if (word !== null) {
-    const symbols = analyzer.getGlobalDeclarationSymbols(textDocumentPositionParams.textDocument.uri)
-
-    // Covert symbols to completion items
-    // TODO: remove duplicate symbols
-    symbolCompletions = symbols.map((symbol: SymbolInformation) => (
-      {
-        label: symbol.name,
-        kind: symbolKindToCompletionKind(symbol.kind),
-        documentation: `${symbol.name}`
-      }
-    ))
-  }
-
-  const allCompletions = [
-    ...contextHandler.getComletionItems(textDocumentPositionParams, documentAsText),
-    ...symbolCompletions
-  ]
-
-  return allCompletions
-})
+connection.onCompletion(onCompletionHandler.bind(this))

```

Comment:
I don't think the `bind(this)` is needed

Reply:
ye, it is not needed for now unless we are going to make server a class